peps/pep-0786.rst

```diff
@@ -0,0 +1,355 @@
+
+..
+  XXX Give a good example of non-hardware aware use of modular arithmetic formatting like Minecraft buried treasure always being at 8,8 within a chunk.
+
+Thus we have defended the lossy behavior of ``z`` implemented as modulo-precision, and we have exhausted all reasonable use cases of lossless behavior.
+
+A final compromise to consider and reject is implementing ``z`` not as a flag *dependent* on ``.``, but as a flag that can be *combined* with ``.``. Specifically: ``z`` without ``.`` would turn on two's complement mode to render the minimal width representation of the formatted integer, ``.`` without ``z`` would implement precision as already explained, a minimum number of digits in the magnitude and a sign if necessary, and ``z`` combined with ``.`` would turn on the left-truncating modulo-precision. This labyrinth of combinations does not seem useful to anyone, as we have already discredited the ergonomics of minimal width representation convention, whence ``z`` would rarely be used on its own, and this behavior of two options that individually render a *minimum* number of digits combining together to render an *exact* number of digits seems counterintuitive.
+
+
+Infinite Length Indication
+''''''''''''''''''''''''''
+
+Another, less popular, rejected alternative was for ``z`` to directly acknowledge the infinite prefix of ``0``\ s or ``1``\ s that precede a non-negative or negative number respectively. For example
+
+.. code-block:: python
+
+    >>> f"{-1:z#.8b}"
+    '0b[...1]11111111'
+    >>> f"{300:z#.8b}"
+    '0b[...0]100101100'
+
+This is effectively the minimal width representation convention with an 'infinite' prefix attached to it.
+
+In the C programming language the machine-width dependent two's complement formatting of ``int`` data with precision exhibits excessive lengths of prefixes that arise from negative numbers, even those with small magnitude:
+
+.. code-block:: C
+
+    printf("%#.2x\n", -19); // 0xffffffed
+    printf("%#.2llx\n", (long long unsigned int)-19); // 0xffffffffffffffed
+
+This prefix could continue on indefinitely if it were not limited by a maximum machine-width!
+
+Python's ``int`` type is indeed not limited by a maximum machine-width. Thus to avoid printing infinitely long two's complement strings we could use a similar approach to that of the builtin ``list``'s string formatting for printing a list that contains itself:
+
+.. code-block:: python
+
+    >>> l = []
+    >>> l.append(l)
+    >>> l
+    [[...]]
+
+    >>> y = -1
+    >>> f"{y:z#.8b}"
+    '0b[...1]11111111'
+
+This may have been useful to educate beginner users on how bitwise binary operations work, for example showing how ``-1 & x`` is always trivially equal to ``x``, or how the binary representation of the negation of a number can be obtained by adding one to its bitwise complement:
```

Comment:
```suggestion
This may have been useful to educate beginners on how bitwise binary operations work, for example showing how ``-1 & x`` is always trivially equal to ``x``, or how the binary representation of the negation of a number can be obtained by adding one to its bitwise complement:
```